src/commands/setup/apply-gradle-plugin/inject-plugin-in-build-gradle.ts

```diff
@@ -28,16 +56,30 @@ export const injectPluginInBuildGradle = async (
         return `${installationBlock}${EOL}${line}`;
       }
 
-      if (line.match("applicationVariants.all") && !hasAddedAutomation) {
-        hasAddedAutomation = true;
-        const automationBlock = [
-          `        if (project.tasks.findByName("minify\${variant.name.capitalize()}WithR8")) {`,
-          `            tasks["minify\${variant.name.capitalize()}WithR8"].finalizedBy { tasks["uploadMapping\${variant.name.capitalize()}"] }`,
-          `        }`,
-          ``,
-        ].join(EOL);
+      // Entering the android closure
+      if (
+        line.match(/^android {/) &&
+        hasAddedAutomationState === AutomationState.BEFORE_ANDROID_CLOSURE
+      ) {
+        hasAddedAutomationState = AutomationState.IN_ANDROID_CLOSURE;
+      }
+
+      // Entering the applicationVariants loop
+      if (
+        line.match("applicationVariants.all") &&
```

Comment:
If I'm not wrong, call to `all` function is composite, meaning you can call it multiple times with different closures and then all the closures will be executed in the sequence.

Given that, even if there is already `applicationVariants.all` call with closure in the file, we can add another call. Will it simplify approach or not?

Change lgtm to me, I'm just wondering if we can do things simpler, without introducing the state.

Reply:
I agree that's a lot simpler and easier to maintain! I'll make the change :)